src/ArcxAnalyticsSdk.ts

```diff
@@ -75,6 +82,24 @@ export class ArcxAnalyticsSdk {
     )
   }
 
+  /*
+    Sent object in eth_sendTransaction is describe under link below:
+    https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendtransaction
+  */
+  private trackTransactions() {
+    const provider = getWeb3Provider()
+    if (!provider) {
+      return
+    }
+    const request = provider.request
+    provider.request = ({ method, params }: RequestArguments) => {
+      if (method === 'eth_sendTransaction') {
+        this.catchedTransaction(Array.isArray(params) ? params[0] : params)
```

Comment:
There's no point in having the `catchedTransaction` function because you can just call `this.transaction()`

Reply:
It doesn't match the signature of `this.transaction()`

Reply:
Inline call